build-docker.sh

```diff
@@ -1,14 +1,42 @@
+#!/usr/bin/env bash
 set -o nounset
 set -o pipefail
 set -o xtrace
 
-TAGS="latest"
+TAGS=""
+# Check if CIRCLE_BRANCH is set and not empty
+if [[ -n "${CIRCLE_BRANCH}" ]]; then
+    export TAGS="${CIRCLE_BUILD_NUM},branch-${CIRCLE_BRANCH}"
+fi
+
+# We only want to tag main with latest on ECR
+if [[ "${CIRCLE_BRANCH}" == "main" ]]; then
+    TAGS+=",latest"
+fi
+
+# Local builds just get "dev" for version
 VERSION="dev"
+
+# If we are doing a dev build on circle, we will version it as the circleci buildnumber
+if [[ -n "${CIRCLE_BUILD_NUM}" ]]; then
```

Comment:
Will CIRCLE_BUILD_NUM ever not be set?  Does this case render `VERSION="dev"` obsolete?

Reply:
I wasn't sure how people do local builds - do those go through circle?

Reply:
You're right, you could technically run this script locally.